bundles/org.openhab.core/src/main/java/org/openhab/core/internal/items/ExpireManager.java

```diff
@@ -0,0 +1,395 @@
+        EventSubscriber.class }, configurationPid = "org.openhab.expire", configurationPolicy = ConfigurationPolicy.OPTIONAL)
+public class ExpireManager implements EventSubscriber, RegistryChangeListener<Item> {
+
+    protected static final String EVENT_SOURCE = "org.openhab.core.expire";
+    protected static final String METADATA_NAMESPACE = "expire";
+    protected static final String PROPERTY_ENABLED = "enabled";
+
+    private static final Set<String> SUBSCRIBED_EVENT_TYPES = Set.of(ItemStateEvent.TYPE, ItemCommandEvent.TYPE);
+
+    private final Logger logger = LoggerFactory.getLogger(ExpireManager.class);
+
+    private Map<String, @Nullable Optional<ExpireConfig>> itemExpireConfig = new ConcurrentHashMap<>();
+    private Map<String, Instant> itemExpireMap = new ConcurrentHashMap<>();
+
+    private ScheduledExecutorService threadPool = ThreadPoolManager
+            .getScheduledPool(ThreadPoolManager.THREAD_POOL_NAME_COMMON);
+
+    private final EventPublisher eventPublisher;
+    private final MetadataRegistry metadataRegistry;
+    private final ItemRegistry itemRegistry;
+    /* default */ final MetadataChangeListener metadataChangeListener = new MetadataChangeListener();
+
+    private boolean enabled = true;
+
+    private @Nullable ScheduledFuture<?> expireJob;
+
+    @Activate
+    public ExpireManager(Map<String, @Nullable Object> configuration, final @Reference EventPublisher eventPublisher,
+            final @Reference MetadataRegistry metadataRegistry, final @Reference ItemRegistry itemRegistry) {
+        this.eventPublisher = eventPublisher;
+        this.metadataRegistry = metadataRegistry;
+        this.itemRegistry = itemRegistry;
+
+        modified(configuration);
+    }
+
+    @Modified
+    protected void modified(Map<String, @Nullable Object> configuration) {
+        Object valueEnabled = configuration.get(PROPERTY_ENABLED);
+        if (valueEnabled != null) {
+            enabled = Boolean.parseBoolean(valueEnabled.toString());
+        }
+        if (enabled) {
+            if (expireJob == null) {
+                expireJob = threadPool.scheduleWithFixedDelay(() -> {
+                    if (!itemExpireMap.isEmpty()) {
+                        for (String itemName : itemExpireConfig.keySet()) {
+                            if (isReadyToExpire(itemName)) {
+                                expire(itemName);
+                            }
+                        }
+                    }
+                }, 1, 1, TimeUnit.SECONDS);
+            }
+            itemRegistry.addRegistryChangeListener(this);
+            metadataRegistry.addRegistryChangeListener(metadataChangeListener);
+        } else {
+            itemRegistry.removeRegistryChangeListener(this);
+            metadataRegistry.removeRegistryChangeListener(metadataChangeListener);
+            if (expireJob != null) {
+                expireJob.cancel(true);
+                expireJob = null;
+            }
```

Comment:
For thread-safty we should use a local variable here.

```suggestion
            ScheduledFuture<?> localExpireJob = expireJob;
            if (localExpireJob != null) {
                localExpireJob.cancel(true);
                expireJob = null;
            }
```